Common/src/toolboxes/C1DInterpolation.cpp

```diff
@@ -0,0 +1,179 @@
+/*!
+ * \file C1DInterpolation.cpp
+ * \brief Inlet_interpolation_functions
+ * \author Aman Baig
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation 
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#include "../../include/toolboxes/C1DInterpolation.hpp"
+
+su2double CAkimaInterpolation::EvaluateSpline(su2double Point_Interp){
+    Point_Match = true;
+    int i =0, j=n-1;
+
+    while (j-i>1){ 
+        int m=(i+j) / 2 ; 
+            if (Point_Interp>x[m]) {i=m;}
+            else {j=m;} 
+    }
+
+    su2double h=Point_Interp-x[i] ;
+    return y[i]+h*(b[i]+h*(c[i]+h*d[i])) ;
+}
+
+su2double CLinearInterpolation::EvaluateSpline(su2double Point_Interp){
+    int size = x.size();
+
+    for (int i=0;i<size-1;i++){
+        if(Point_Interp>=x[i] && Point_Interp<=x[i+1]){
+            Point_Match = true;
+            return (Point_Interp-x[i])*dydx[i]+y[i];}
+    }
+    return 0;
+}
```

Comment:
You check "PointMatch" after evaluating the spline but I don't see here where that boolean is set to false in case of failure.

Reply:
The default set in C1DInterpolation class protected variables: `bool Point_Match = false`

Reply:
I had set the default value in the initialization of C1DInterpolation class protected variables. But I changed that now to keep it default as false for every call of the evaluation function. I will push the new commit soon.